tox.ini

```diff
@@ -37,14 +37,7 @@ passenv =
     dev: JSTOR_API_URL
     dev: JSTOR_API_SECRET
 deps =
-    dev: -r requirements/dev.txt
-    {format,checkformatting}: -r requirements/format.txt
-    lint: -r requirements/lint.txt
-    tests: -r requirements/tests.txt
-    coverage: -r requirements/coverage.txt
-    functests: -r requirements/functests.txt
-    bddtests: -r requirements/bddtests.txt
-    dockercompose: -r requirements/dockercompose.txt
+    -r requirements/{env:TOX_ENV_NAME}.txt
```

Comment:
No functional change here I just collapsed the several separate `deps` lines into one by using the `TOX_ENV_NAME` envvar (and relying on the `checkformatting.txt` -> `format.txt` symlink that this PR adds)

Reply:
I think just biting the bullet and having a matching name is one of the things we should have been doing all along to reduce edge cases and complexity.

Reply:
Yeah, avoiding special cases is a good idea.

This is something I've realised over time. The way that tox naturally wants to work is one requirements file pair and corresponding venv per "command": `tox -e lint` has `requirements/lint.{in,txt}` and `.tox/lint/`, etc.

More than once we've attempted to deviate from this. For example it has often been pointed out as "silly" that there's a `.tox/coverage/` venv with just `coverage` installed in it (for running the `coverage report` command) when `coverage` already has to be installed in the `tests` env in order to run the tests with coverage collection. But this is not silly:

* It means that to print out the coverage report you just do `tox -e coverage` instead of some painful long command like `tox -e tests --run-command 'coverage report'`
* It means that we can inform tox of the dependency of the `coverage` env on the `tests` env so that if you run `tox --parallel -e checkformatting,lint,tests-py{38,39,310},coverage,functests` tox will run everything in parallel but will know not to begin the coverage report until the unit tests for all versions of Python are finished (`make sure` does this)

* So there's a "pointless" `coverage.{in,txt}` file pair and `.tox/coverage/` venv. It's not in fact pointless at all. And what harm does it cause? Are you running out of hard drive space? (The `coverage` venv is 21MB and does not take a significant amount of time to create.)

Similarly we wanted to avoid having both `format.{in,txt}` and `checkformatting.{in,txt}` even though we did have separate `.tox/format/` and `.tox/checkformatting/` venvs. All that did was make things awkward.

Let tox be tox